target/linux/rtl838x/files-5.4/arch/mips/rtl838x/serial.c

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * 8250 serial console setup for the Realtek RTL838X SoC
+ *
+ * based on the original BSP by
+ * Copyright (C) 2006-2012 Tony Wu (tonywu@realtek.com)
+ *
+ * Copyright (C) 2020 B. Koblitz
+ *
+ */
+#include <linux/types.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/version.h>
+#include <linux/serial.h>
+#include <linux/serial_core.h>
+#include <linux/serial_8250.h>
+#include <linux/serial_reg.h>
+#include <linux/tty.h>
+#include <linux/clk.h>
+
+#include "mach-rtl838x.h"
+
+extern char arcs_cmdline[];
+
+int __init rtl838x_serial_init(void)
+{
+#ifdef CONFIG_SERIAL_8250
+	int ret;
+	struct uart_port p;
+	int baud = 0;
+	char parity = '\0', bits = '\0', flow = '\0';
+	char *s, *n;
+
+	/* Enable UART1 */
+	rtl838x_w32(0x10, RTL838X_GMII_INTF_SEL);
+
+	s = strstr(arcs_cmdline, "console=ttyS0,");
+	if (s) {
+		s += 14;
+		baud = simple_strtoul(s, &n, 10);
+		s = n;
+		if (*s == ',')
+			s++;
+		if (*s)
+			parity = *s++;
+		if (*s == ',')
+			s++;
+		if (*s)
+			bits = *s++;
+		if (*s == ',')
+			s++;
+		if (*s == 'h')
+			flow = 'r';
```

Comment:
It looks like flow is only written ad never read.

Reply:
I will remove that. Not sure the SoC does actually do flow control...